plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,381 @@
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    description: My Ansible Port Channel
+    port_channel_interface: ansible_port_channel_interface
+    node: 101
+    interfaces:
+      - 1/1
+      - 1/10-11
+    interface_policy_uuid: ansible_port_channel_policy
+    interface_descriptions:
+      - interface_id: 1/1
+        description: My first Ansible Interface
+      - interface_id: 1/10
+        description: My second Ansible Interface
+      - interface_id: 1/11
+        description: My third Ansible Interface
+    state: present
+
+- name: Query a Port Channel Interface with template_name
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: query
+  register: query_one
+
+- name: Query all Port Channel Interfaces in the template
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete a Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+    lookup_valid_interfaces,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        port_channel_interface=dict(type="str", aliases=["name", "port_channel"]),
+        port_channel_interface_uuid=dict(type="str", aliases=["uuid", "port_channel_uuid"]),
+        description=dict(type="str"),
+        node=dict(type="str"),
+        interfaces=dict(type="list", elements="str", aliases=["members"]),
+        interface_policy=dict(type="str", aliases=["policy", "pc_policy"]),
+        interface_policy_uuid=dict(type="str", aliases=["policy_uuid", "pc_policy_uuid"]),
+        interface_descriptions=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                interface_id=dict(type="str"),
+                description=dict(type="str"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["template", "port_channel_interface"]],
```

Comment:
template is already required in line 171, also add port_channel_interface_uuid in state